cpp/tensorrt_llm/kernels/groupRmsNormKernels/groupRmsNormKernels.cu

```diff
@@ -654,4 +704,95 @@ void GroupRMSNormKernelLargeBatchLauncher(GroupRMSParams<n> params)
 
 INSTANTIATE_GROUP_RMS_NORM_LARGE_BATCH(2)
 
+int getComputeCapabilityMajor()
+{
+    int device;
+    TLLM_CUDA_CHECK(cudaGetDevice(&device));
+    cudaDeviceProp prop;
+    TLLM_CUDA_CHECK(cudaGetDeviceProperties(&prop, device));
+    return prop.major;
+}
+
+bool prefer_base_kernel(int batch, int base_warps, float scheduling_efficiency_ratio)
+{
+    int sm_major = getComputeCapabilityMajor();
+    for (auto const& [known_model, model] : gpu_models)
+    {
+        if (sm_major == known_model)
+        {
+            float p = model.batch_size * batch + model.base_warps * base_warps
+                + model.scheduling_efficiency_ratio * scheduling_efficiency_ratio + model.intercept;
+            p = 1.0f / (1.0f + std::exp(-p));
+            return p > 0.5f;
+        }
+    }
+    return true;
+}
+
+template <int n>
+void GroupRMSNormKernelLauncherWithHeuristic(GroupRMSParams<n> params)
+{
+    if (params.num_inputs == 1)
+    {
+        GroupRMSNormBaseKernelLauncher<n>(params);
+    }
+    else if (params.num_inputs == 2)
+    {
+        int num_warps_per_sm = 64;
+        uint32_t base_warps;
+        uint32_t large_batch_warps;
+
+        // Choose the appropriate DType
+        switch (params.dtype)
+        {
+        case nvinfer1::DataType::kHALF:
+            base_warps = calculateNumWarpsBase<half, n>(params);
+            large_batch_warps = calculateNumWarpsLargeBatch<half, n>(params).num_warps_to_launch;
+            break;
+        case nvinfer1::DataType::kBF16:
+            base_warps = calculateNumWarpsBase<__nv_bfloat16, n>(params);
+            large_batch_warps = calculateNumWarpsLargeBatch<__nv_bfloat16, n>(params).num_warps_to_launch;
+            break;
+        case nvinfer1::DataType::kFLOAT:
+            base_warps = calculateNumWarpsBase<float, n>(params);
+            large_batch_warps = calculateNumWarpsLargeBatch<float, n>(params).num_warps_to_launch;
+            break;
+        default: TLLM_CHECK_WITH_INFO(false, "Unsupported data type for GroupRMSNorm"); return;
+        }
+
+        int block_limit_warps_base = std::floor(num_warps_per_sm / base_warps);
+        int block_limit_warps_large_batch = std::floor(num_warps_per_sm / large_batch_warps);
```

Comment:
`block_limit_warps` is a bit confusing. Can you rename it?

Reply:
Renamed to `concurrent_block_per_sm_xxx`.